src/views/Rules/ListRules/index.jsx

```diff
@@ -222,23 +243,73 @@ function ListRules(props) {
       rowsPerPage={RULES_ROWS_PER_PAGE}
     />
   );
+  const handleDateTimePickerError = error => {
+    setDateTimePickerError(error);
+  };
+
+  const handleDateTimeChange = date => {
+    setScheduleDeleteDate(date);
+    setDateTimePickerError(null);
+  };
+
+  const handleDialogError = error => {
+    setDialogState({ ...dialogState, error });
+  };
+
+  const dialogBody =
+    dialogState.item &&
+    (Object.keys(dialogState.item.requiredSignoffs).length > 0 ? (
+      <DateTimePicker
+        disablePast
+        inputVariant="outlined"
+        fullWidth
+        label="When"
+        onError={handleDateTimePickerError}
+        helperText={
+          dateTimePickerError ||
+          (scheduleDeleteDate < new Date() ? 'Scheduled for ASAP' : undefined)
+        }
+        onDateTimeChange={handleDateTimeChange}
+        value={scheduleDeleteDate}
+      />
+    ) : (
+      `This will delete rule ${dialogState.item.rule_id}.`
+    ));
   const handleRuleDelete = rule => {
     setDialogState({
       ...dialogState,
       open: true,
       title: 'Delete Rule?',
-      body: `This will delete rule ${rule.rule_id}.`,
       confirmText: 'Delete',
       item: rule,
     });
   };
 
-  const handleDialogError = error => {
-    setDialogState({ ...dialogState, error });
-  };
-
-  const handleDialogClose = () => {
+  const handleDialogClose = result => {
```

Comment:
`onClose` is triggered when the user clicks cancel or the ESC key. `handleDialogClose` will be called since we have`onClose={handleDialogClose}`. This might cause an issue since no result will be passed as a param. We probably want to have a `handleDialogComplete` handler that does the logic then calls `handleDialogClose` at the end. Somthing like:


```jsx
const handleDialogComplete = result => {
  // handle logic here

  this.handleDialogClose();
};

// ...
onComplete={handleDialogComplete}
onClose={handleDialogClose}
```
`handleDialogClose` also gets called when the use